thirdparty/superchain.go

```diff
@@ -0,0 +1,341 @@
+		if strings.Contains(spec, "raw.githubusercontent.com") {
+			if strings.HasPrefix(spec, "http://") || strings.HasPrefix(spec, "https://") {
+				return spec, nil
+			}
+			// Should ideally not happen: a raw URL without a scheme. Prepend https.
+			return "https://" + spec, nil
+		}
+
+		var pathPart string
+		if strings.HasPrefix(spec, "https://github.com/") {
+			pathPart = strings.TrimPrefix(spec, "https://github.com/")
+		} else if strings.HasPrefix(spec, "http://github.com/") {
+			pathPart = strings.TrimPrefix(spec, "http://github.com/")
+		} else { // must be "github.com/"
+			pathPart = strings.TrimPrefix(spec, "github.com/")
+		}
+
+		parts := strings.Split(strings.Trim(pathPart, "/"), "/")
+		if len(parts) < 2 {
+			return "", fmt.Errorf("invalid GitHub superchain spec: '%s' (org/repo not found after prefix)", spec)
+		}
+
+		org, repo := parts[0], parts[1]
+		branch := "main"
+		jsonFile := "chainList.json"
+
+		if len(parts) > 2 { // Potential branch or full path
+			// Check if parts[2] looks like a common branch name or if it's part of a longer path ending in .json
+			// This logic assumes if more than org/repo is given, it might include branch and/or filename.
+			// Example: org/repo/my-branch
+			// Example: org/repo/my-branch/customList.json
+			// Example: org/repo/main/some/dir/customList.json (less common for this use case)
+			if len(parts) == 3 && !strings.HasSuffix(parts[2], ".json") { // org/repo/branch
+				branch = parts[2]
+			} else if len(parts) >= 3 { // org/repo/branch/file.json or org/repo/file.json (implicit main)
+				// Determine if parts[2] is a branch or part of the filename
+				// If parts[2] is not a .json file, assume it's a branch
+				if !strings.HasSuffix(parts[2], ".json") {
+					branch = parts[2]
+					if len(parts) > 3 {
+						jsonFile = strings.Join(parts[3:], "/")
+					}
+				} else {
+					// This means spec was like github.com/org/repo/somefile.json, assume main branch
+					jsonFile = strings.Join(parts[2:], "/")
+				}
+			}
+		}
+		return fmt.Sprintf("https://raw.githubusercontent.com/%s/%s/%s/%s", org, repo, branch, jsonFile), nil
+	}
+
+	// Already a full URL (e.g. raw.githubusercontent.com or other custom registry)?
+	if strings.HasPrefix(spec, "http://") || strings.HasPrefix(spec, "https://") {
+		return spec, nil
+	}
+
+	// Fallback: treat as a domain/path and prepend https scheme.
+	return "https://" + spec, nil
+}
+
+type SuperchainNetwork struct {
+	Name                 string   `json:"name"`
+	Identifier           string   `json:"identifier"`
+	ChainID              int64    `json:"chainId"`
+	RPC                  []string `json:"rpc"`
+	Explorers            []string `json:"explorers"`
+	SuperchainLevel      int      `json:"superchainLevel"`
+	GovernedByOptimism   bool     `json:"governedByOptimism"`
+	DataAvailabilityType string   `json:"dataAvailabilityType"`
+	Parent               struct {
+		Type  string `json:"type"`
+		Chain string `json:"chain"`
+	} `json:"parent"`
```

Comment:
If we're not using other fields no need to parse the, if we only use "rpc" probably let's just use map[string]?